org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/synapse/schemagen/SchemaGeneratorForCSVTest.java

```diff
@@ -0,0 +1,203 @@
+package org.eclipse.lemminx.synapse.schemagen;
+
+import org.eclipse.lemminx.customservice.synapse.schemagen.util.FileType;
+import org.eclipse.lemminx.customservice.synapse.schemagen.util.SchemaGeneratorForCSV;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class SchemaGeneratorForCSVTest {
+
+    private void verifyGeneratedJSONSchema(String csvFilePath, String jsonSchemaFilePath, String delimiter)
+            throws IOException {
+
+        String generatedSchema = generateJSONSchema(csvFilePath, delimiter);
+        String expectedSchema = new String(Files.readAllBytes(Paths.get(jsonSchemaFilePath)));
+        assertEquals(expectedSchema.replace("\r\n", "").replace("\n", "").replace(" ", "").trim(),
+                generatedSchema.trim().replace(" ", "").trim());
+    }
+
+    private String generateJSONSchema(String csvFilePath, String delimiter) throws IOException {
+
+        SchemaGeneratorForCSV schemaGenerator = new SchemaGeneratorForCSV();
+        String fileContent = new String(Files.readAllBytes(Paths.get(csvFilePath)));
+        return schemaGenerator.getSchemaContent(fileContent, FileType.CSV, delimiter);
+    }
+
+    @Test
+    public void testGenerateSchemaWithBasicWithHeaders() throws IOException {
+
+        String csvFilePath = "src/test/resources/synapse/schemagen/csv/sampleWithBasicWithHeaders.csv";
+        String expectedJSONSchemaFilePath =
+                "src/test/resources/synapse/schemagen/csv/expectedJsonSchema/sampleWithBasicWithHeaders.json";
+        verifyGeneratedJSONSchema(csvFilePath, expectedJSONSchemaFilePath, ",");
```

Comment:
Can we add two parameters(csvFilePath, expectedJSONSchemaFilePath) to the `verifyGeneratedJSONSchema` method? If we do, we can remove the path initialization.